crates/l2/sequencer/state_diff.rs

```diff
@@ -183,4 +320,168 @@ impl AccountStateDiff {
 
         Ok((r#type, Bytes::from(encoded)))
     }
+
+    /// Returns a tuple of the number of bytes read, the address of the account
+    /// and the decoded `AccountStateDiff`
+    pub fn decode(bytes: &[u8]) -> Result<(usize, Address, Self), StateDiffError> {
+        let mut decoder = Decoder::new(bytes);
+
+        let update_type = decoder.get_u8()?;
+
+        let address = decoder.get_address()?;
+
+        let new_balance = if update_type.is(AccountStateDiffType::NewBalance) {
+            Some(decoder.get_u256()?)
+        } else {
+            None
+        };
+
+        let nonce_diff = if update_type.is(AccountStateDiffType::NonceDiff) {
+            Some(decoder.get_u16()?)
+        } else {
+            None
+        };
```

Comment:
Tip: you can use [then_some](https://doc.rust-lang.org/std/primitive.bool.html#method.then_some) for brevity:
```suggestion
        let new_balance = update_type.is(AccountStateDiffType::NewBalance)
            .then_some(decoder.get_u256()?);
        let nonce_diff = update_type.is(AccountStateDiffType::NonceDiff)
            then_some(decoder.get_u16()?);
```